File: src/net/dodian/uber/game/model/player/casino/Spin.java
package net.dodian.uber.game.model.player.casino;

import net.dodian.uber.game.Server;

public class Spin {
  Symbol[] symbols = new Symbol[3];
  int winnings = 0;
  
  public Spin(Symbol[] symbols) {
	    this.symbols[0] = symbols[0];
	    this.symbols[1] = symbols[1];
	    this.symbols[2] = symbols[2];
	    symbols[0].setColor("red");
	    symbols[1].setColor("red");
	    symbols[2].setColor("red");
	    if (symbols[0].getId() == symbols[1].getId() && symbols[1].getId() == symbols[2].getId()) {
    	    symbols[0].setColor("gre");
    	    symbols[1].setColor("gre");
    	    symbols[2].setColor("gre");
	    	if(symbols[0].getId() == 8) //Jackpot!
	    	    winnings = Server.slots.slotsJackpot + Server.slots.peteBalance >= Integer.MAX_VALUE ? Integer.MAX_VALUE : Server.slots.slotsJackpot + Server.slots.peteBalance;
	    	else if(symbols[0].getId() == 7)
	    	    winnings = 25000;
	    	else
	    		winnings = 3500;
	    } else if (symbols[0].getId() == symbols[1].getId() || symbols[1].getId() == symbols[2].getId() || symbols[0].getId() == symbols[2].getId()) {
	        int id = -1;
	        if (symbols[0] == symbols[1]) {
	          id = symbols[0].getId();
	    	  symbols[0].setColor("gre");
	    	  symbols[1].setColor("gre");
	        } else if (symbols[0] == symbols[2]) {
	          id = symbols[0].getId();
	    	  symbols[0].setColor("gre");
	    	  symbols[2].setColor("gre");
	        } else if (symbols[1] == symbols[2]) {
	          id = symbols[1].getId();
	    	  symbols[1].setColor("gre");
	    	  symbols[2].setColor("gre");
	        }
	        if(id != -1) {
	        	if(id == 8)
	        		winnings = 30000;
	        	else if(id == 7)
	        		winnings = 11000;
	        	else
	        		winnings = 1500;
	        }
	    } else {
	    	boolean found = false;
	    	for(int i = 0; i < symbols.length && !found; i++)
	    		if(symbols[i].getId() == 8 || symbols[i].getId() == 7) {
	    			symbols[i].setColor("gre");
	    			winnings = 1500;
	    			found = true;
	    		} else winnings = 0;
	    }
	  }

  public Symbol[] getSymbols() {
    return symbols;
  }

  public int getWinnings() {
    return winnings;
  }
}

File: game-server/src/main/kotlin/net/dodian/uber/game/model/player/casino/Spin.kt
package net.dodian.uber.game.model.player.casino

import net.dodian.uber.game.Server

class Spin(
    private val symbols: Array<Symbol>,
) {
    private var winnings: Int = 0

    init {
        symbols[0].setColor("red")
        symbols[1].setColor("red")
        symbols[2].setColor("red")

        if (symbols[0].getId() == symbols[1].getId() && symbols[1].getId() == symbols[2].getId()) {
            symbols[0].setColor("gre")
            symbols[1].setColor("gre")
            symbols[2].setColor("gre")
            winnings =
                when (symbols[0].getId()) {
                    8 -> {
                        val total = Server.slots.slotsJackpot + Server.slots.peteBalance
                        if (total >= Int.MAX_VALUE) Int.MAX_VALUE else total
                    }

                    7 -> 25_000
                    else -> 3_500
                }
        } else if (symbols[0].getId() == symbols[1].getId() || symbols[1].getId() == symbols[2].getId() || symbols[0].getId() == symbols[2].getId()) {
            var id = -1
            if (symbols[0].getId() == symbols[1].getId()) {
                id = symbols[0].getId()
                symbols[0].setColor("gre")
                symbols[1].setColor("gre")
            } else if (symbols[0].getId() == symbols[2].getId()) {
                id = symbols[0].getId()
                symbols[0].setColor("gre")
                symbols[2].setColor("gre")
            } else if (symbols[1].getId() == symbols[2].getId()) {
                id = symbols[1].getId()
                symbols[1].setColor("gre")
                symbols[2].setColor("gre")
            }
            winnings =
                when (id) {
                    8 -> 30_000
                    7 -> 11_000
                    -1 -> 0
                    else -> 1_500
                }
        } else {
            var found = false
            for (symbol in symbols) {
                if (symbol.getId() == 8 || symbol.getId() == 7) {
                    symbol.setColor("gre")
                    winnings = 1_500
                    found = true
                    break
                }
            }
            if (!found) {
                winnings = 0
            }
        }
    }

    fun getSymbols(): Array<Symbol> = symbols

    fun getWinnings(): Int = winnings
}
